File: actividad1.java
// Este es un comentario de una sola línea

/*
   Este es un comentario multilínea o de bloque
   Útil para explicaciones detalladas
*/

/**
 * Este es un comentario de documentación (Javadoc)
 * Es exclusivo de Java y se utiliza para generar documentación HTML 
 * automática de nuestro código mediante herramientas del sistema
 */

//Alumna: Lucero Villarreal Sotero
//Profesor: Jaciel Luna
//Módulo: Programación

/*
ej01- Crea un comentario en el código y coloca el nombre de la asignatura 
- Representa las diferentes sintaxis que existen de crear comentarios en el lenguaje (en una línea, varias...). Escribe el nombre del profesor y tus datos generales
- Crea una variable (y una constante si el lenguaje lo soporta). 
- Crea variables representando todos los tipos de datos primitivos del lenguaje (cadenas de texto, enteros, booleanos...). 
- Imprime por terminal el texto: "¡Hola, [y el nombre del lenguaje]!"
*/


/*---------------------------------------------------
Sintaxis
---------------------------------------------------*/

//En Java, todo el código debe estar dentro de una clase (class)
//El nombre de la clase debe coincidir exactamente con el nombre del archivo
public class actividad1 {

// El método "main" es el punto de partida obligatorio de cualquier aplicación Java
// public: accesible desde cualquier lado
// static: no requiere crear un objeto para ejecutarse
// void: no devuelve ningún valor
// String[] args: permite recibir argumentos externos (parámetros) desde la terminal
    public static void main(String[] args) {



/*----------------------------------------------
Declaración de variables y tipos de datos
----------------------------------------------*/     
// Java es un lenguaje de tipado fuerte. Debes declarar el tipo de variable antes de su nombre

//Enteros
    byte numeroMuyPequeño = 127;        // 1 byte, tiene un rango de -128 a 127
    short numeroPequeño = 32767;        // 2 bytes, tiene un rango de -32,768 a 32,767
    int edad = 28;      // 4 bytes, el más utilizado para enteros cotidianos
    long numeroGigante = 9223372036854775807L;      // 8 bytes, requiere una 'L' al final para indicar que es Long

//Decimales
    float pi = 3.14159f;        // 4 bytes, requiere una 'f' al final
    double constanteGravedad = 9.80665;

//Caracteres (un solo símbolo)
    char inicialNombre = 'J';       //Siempre entre comillas simples (' ')

//Booleanos (Verdadero o Falso)
    boolean meGustaProgramar = true;      //Guarda true o false, no se puede imprimir como 1 o 0 directamente

//Cadenas de texto (Clase String)
// Nota: String en Java no es un tipo de dato primitivo, es una clase (por eso empieza con mayúscula)
    String nombreEstudiante = "Juan Pérez"; //Siempre entre comillas dobles (" ")


/*-----------------------------------------------
Imprimir
-----------------------------------------------*/
// System.out.println imprime el texto en la consola y automáticamente hace un salto de línea.
    System.out.println("¡Hola! Bienvenido al mundo de Java.");




/*-----------------------------------------------
Imprimir variales
-----------------------------------------------*/
// En Java concatenamos (unimos) textos y variables usando el operador de suma (+).
    System.out.println("Nombre del estudiante: " + nombreEstudiante);
    System.out.println("Inicial del nombre: " + inicialNombre);
    System.out.println("Edad: " + edad + " años");
    System.out.println("Número Byte: " + numeroMuyPequeño);
    System.out.println("Número Short: " + numeroPequeño);
    System.out.println("Número Long: " + numeroGigante);
    System.out.println("Número Float (pi): " + pi);
    System.out.println("Número Double (gravedad): " + constanteGravedad);
        
// Al imprimir un booleano en Java, la consola mostrará literalmente "true" o "false".
    System.out.println("¿Le gusta programar?: " + meGustaProgramar);

    }
}

File: actividad1.js
// Este es un comentario de una sola línea

/*
   Este es un comentario multilínea o de bloque
   Sirve para escribir explicaciones largas o documentar lógica
*/

/*
 * Este es un comentario de documentación (JSDoc)
 * Es muy popular en el ecosistema de JavaScript para documentar funciones,
 * parámetros y tipos, ayudando a que los editores modernos autocompleten mejor
 */

//Alumna: Lucero Villarreal Sotero
//Profesor: Jaciel Luna
//Módulo: Programación

/*ej01- Crea un comentario en el código y coloca el nombre de la asignatura 
- Representa las diferentes sintaxis que existen de crear comentarios en el lenguaje (en una línea, varias...). Escribe el nombre del profesor y tus datos generales
- Crea una variable (y una constante si el lenguaje lo soporta). 
- Crea variables representando todos los tipos de datos primitivos del lenguaje (cadenas de texto, enteros, booleanos...). 
- Imprime por terminal el texto: "¡Hola, [y el nombre del lenguaje]!"


/*-------------------------------
Sintaxis
-------------------------------*/

/*
En JavaScript moderno no necesitas obligatoriamente envolver tu código en clases o funciones
El código se puede ejecutar de arriba hacia abajo directamente
El uso del punto y coma (;) al final de las líneas es opcional en la mayoría de los casos, pero se considera una buena práctica para evitar comportamientos inesperados
*/

/* ¿Cómo se manejan las Librerías en JavaScript?
   Dependiendo del entorno (Navegador o Node.js/Servidor), existen dos formas comunes de importar librerías:

   1. Método tradicional / Servidor (CommonJS):
      const fs = require('fs'); // Importa la librería nativa para manejar archivos.

   2. Método moderno (ES Modules - frontend y backend moderno):
      import { miFuncion } from './miLibreria.js';
*/



/*----------------------------
DEclaración de variales
----------------------------*/
// En JavaScript existen tres palabras clave para declarar variables:
// A. 'var' (Antiguo): Define variables con ámbito de función. Ya NO se recomienda usar 
// porque puede causar errores de lógica (como la re-declaración silenciosa)
var variableAntigua = "No me uses a menos que sea necesario";

//'let' (Moderno): Define variables con ámbito de bloque (solo existen dentro de las llaves donde se crearon)
let edad = 28;
edad = 29; //El valor cambia de 28 a 29

//'const' (Moderno): Define constantes, no se puede reasignar su valor una vez definido
const pi = 3.14159;
// pi = 3.14; // ¡Esto lanzaría un error! Porque es una constante


//En JS las variables son dinámicas; se adaptan al valor que les asignes en el momento
//Tipos primitivos
//String (Cadenas de texto): Se pueden definir con comillas dobles, simples o backticks (comillas invertidas)
const textoSencillo = "Hola Mundo";
const textoConComillasSimples = 'JavaScript es genial';
// Los backticks (``) permiten interpolar (insertar) variables directamente sin concatenar con '+'
const saludoEspecial = `Mensaje: ${textoSencillo} - Detalle: ${textoConComillasSimples}`;

//Number: en JS todos los números (enteros o decimales) pertenecen al tipo 'Number'
const entero = 42;
const decimal = 99.99;

//BigInt: Para números enteros extremadamente grandes que superen el límite seguro de 'Number'
//Se crean agregando una 'n' al final del número
const numeroMasivo = 90071992547409912345n;

//Boolean: Verdadero o Falso.
const programarEsDivertido = true;
const tengoHambre = false;

//Undefined: significa que la variable ha sido declarada pero aún no tiene ningún valor asignado
let variableVacia; // Por defecto su valor es 'undefined'

//Null: representa la ausencia intencional de cualquier valor u objeto
const valorNulo = null;

//Symbol: crea un identificador único e inmutable (se usa para propiedades avanzadas de objetos)
const miSimbolo = Symbol("id_unico");


//Tipos de referencias (Estructuras de datos complejas)

//Objetos (Object): colecciones de propiedades (clave-valor)
const estudiante = {
    nombre: "Juan Pérez",
    edad: 28,
    estaActivo: true
};

//Arreglos (Array): listas ordenadas de datos (pueden mezclar diferentes tipos)
const lenguajesFavoritos = ["JavaScript", "Python", "C++", 100, true];

//Funciones (Function): bloques de código reutilizables, en JS, las funciones son tratadas como tipos de datos
const saludar = function(nombre) {
    return `Hola, ${nombre}!`;
};



/*----------------------------
Impresión
----------------------------*/
// Para imprimir texto o datos en la terminal (consola) de JavaScript, usamos console.log().
console.log("¡Hola! Bienvenido al dinámico mundo de JavaScript.");



/*----------------------------
Imprimir variales
----------------------------*/
console.log("--- Impresión de Variables ---");
console.log("Texto con interpolación:", saludoEspecial);
console.log("Número Entero:", entero);
console.log("Número Decimal:", decimal);
console.log("Número Gigante (BigInt):", numeroMasivo);
console.log("Booleano:", programarEsDivertido);
console.log("Variable sin definir (Undefined):", variableVacia);
console.log("Valor nulo (Null):", valorNulo);
console.log("Símbolo único:", miSimbolo.toString());
console.log("\n--- Impresión de Estructuras Complejas ---");
console.log("Objeto Estudiante:", estudiante);
console.log("Acceder a una propiedad del objeto (nombre):", estudiante.nombre);
console.log("Lista de Lenguajes (Array):", lenguajesFavoritos);
console.log("Acceder al primer elemento del Array:", lenguajesFavoritos[0]);
console.log("Llamada a la función saludar():", saludar("Carlos"));